### PR TITLE
Add how-to for configuring XFF handling

### PR DESCRIPTION
**Pull request by HappyTetrahedron**

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+== Possible behaviours
+
+* *`append`*: By default, the ingress is configured to append the headers to any existing headers. If `Forwarded` and `X-Forwarded-For` headers are already present when the ingress receives a request, the existing values are preserved, and new values are appended by the ingress.
+* *`replace`*: If forwarded header handling is set to `replace`, any existing forwarded headers are discarded. Only the headers set by the ingress will be available.
+* *`never`*: If forwarded header handling is set to `never`, the ingress will never set these headers, and leave any existing headers untouched.
+* *`if-none`*: If forwarded header handling is set to `if-none`, the ingress will only set the headers if they're not already present. If any headers are present, they're left unchanged.
+
+== Annotating the route
+
+Forwarded header handling is configured by annotating the `Route` object.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc:14]: Probably makes sense to note that the annotations can also be set on `Ingress` objects which are managed by the ingresstoroute controller

File: docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
+= Configuring X-Forwarded-For handling
+
+This how-to explains how to configure the ingress' handling of the forwarded headers (`Forwarded` and `X-Forwarded-For`) on a per-route basis.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc:3]:
```suggestion
This how-to explains how to customize how the forwarding headers (`Forwarded` and `X-Forwarded-For`) are handled by the OpenShift ingress router on a per-route basis. 
```

File: docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
+= Configuring X-Forwarded-For handling
+
+This how-to explains how to configure the ingress' handling of the forwarded headers (`Forwarded` and `X-Forwarded-For`) on a per-route basis. 
+
+== Possible behaviours
+
+* *`append`*: By default, the ingress is configured to append the headers to any existing headers. If `Forwarded` and `X-Forwarded-For` headers are already present when the ingress receives a request, the existing values are preserved, and new values are appended by the ingress.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc:7]:
```suggestion
* *`append`*: By default, the OpenShift router is configured to append the headers to any existing headers.
If `Forwarded` and `X-Forwarded-For` headers are already present when the OpenShift router receives a request, the existing values are preserved, and new values are appended by the router.
```

File: docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+* *`replace`*: If forwarded header handling is set to `replace`, any existing forwarded headers are discarded. Only the headers set by the ingress will be available.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc:8]:
```suggestion
* *`replace`*: If forwarded header handling is set to `replace`, any existing forwarded headers are discarded.
Only the headers set by the OpenShift router will be available.
```

File: docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc (hunk 3 above)
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc:7]: Consider using the following formatting:

```suggestion
`append`:: By default, the ingress is configured to append the headers to any existing headers. If `Forwarded` and `X-Forwarded-For` headers are already present when the ingress receives a request, the existing values are preserved, and new values are appended by the ingress.
```

File: docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+* *`never`*: If forwarded header handling is set to `never`, the ingress will never set these headers, and leave any existing headers untouched.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc:9]:
```suggestion
* *`never`*: If forwarded header handling is set to `never`, the OpenShift router will never set these headers, and leave any existing headers untouched.
```

File: docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+* *`if-none`*: If forwarded header handling is set to `if-none`, the ingress will only set the headers if they're not already present. If any headers are present, they're left unchanged.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc:10]:
```suggestion
* *`if-none`*: If forwarded header handling is set to `if-none`, the OpenShift router will only set the headers if they're not already present.
If any headers are present, they're left unchanged.
```

**Review by simu — CHANGES_REQUESTED** [review]
This page should be part of https://github.com/appuio/managed-openshift-docs (the end-user docs for APPUiO Managed OCP4) and not kb.vshn.ch/oc4.

**Comment by HappyTetrahedron** [comment]
Closed in favor of https://github.com/appuio/managed-openshift-docs/pull/7

**Pull request closed without merging**